src/modules/sway/bar.cpp

```diff
@@ -4,12 +4,30 @@
 #include <spdlog/spdlog.h>
 
 #include <stdexcept>
+#include <unordered_set>
 
 #include "bar.hpp"
 #include "modules/sway/ipc/ipc.hpp"
 
 namespace waybar::modules::sway {
 
+void checkModuleActive(const Json::Value& config, std::map<std::string, bool>& module_check) {
+  std::unordered_set<std::string> active_modules;
+  for (const auto& section : {"modules-left", "modules-center", "modules-right"}) {
+    if (const auto& modules = config.get(section, {}); modules.isArray()) {
+      for (const auto& module : modules) {
+        active_modules.insert(module.asString());
+      }
+    }
+  }
+
+  for (auto& module : module_check) {
+    if (active_modules.find(module.first) != active_modules.end()) {
```

Comment:
Note that `sway/workspaces#bar-0` with an optional id after `#` is a valid module reference in the config. Exact string comparison will fail to find that, thus the prefix match with `module.asString().rfind(name,0)` in my PoC code.

To address that, you can try dropping the module ids (`#` and everything after) at line 19.

Reply:
Your PoC code was cleaner than the one I added now. Any preference on how to do it differently? My thinking was to not do the loop multiple times...

Reply:
Well, that's a short loop (just how many modules can be placed on a single bar?) and it's done only once for each visible bar instance. I'd personally pick readability over the optimization here.

Reply:
Made the function into a member, looked nicer with the config access, hope that is not a problem?